site/src/pages/docs/advanced-streaming-backpressure.mdx

```diff
@@ -0,0 +1,124 @@
+# Sending streaming responses using backpressure
+
+When you design a service that sends an <type://HttpResponse> whose payload is large, you should consider how
+you load data into memory. Otherwise, you will encounter an `OutOfMemoryError` when the service serves
+multiple requests at the same time.
+
+One of the options to solve the issue is splitting the payload into chunks and loading them one by one
+into memory. Then, the API sends the chunks one by one as a streaming response using backpressure.
+Backpressure basically means that the API sends the streaming response when it's able to send and when
+the client is ready to receive the responses.
+
+<Tip>
+
+See [Let’s Play with Reactive Streams on Armeria - 1](https://engineering.linecorp.com/en/blog/reactive-streams-armeria-1/)
+to understand backpressure and the situation when an `OutOfMemoryError` is raised.
+
+</Tip>
+
+Let's look at how we can create a streaming <type://HttpResponse> before we get to the backpressure.
+You can create an <type://HttpResponseWriter> from <type://HttpResponse#streaming()> and write
+<type://ResponseHeaders> and chunks in order:
```

Comment:
I guess most users don't want to directly use the `HttpResponseWriter` API.
How about introducing a simpler API first and explain `HttpResponseWriter` later in this documentation.
For example:
```java
import com.linecorp.armeria.common.HttpData;
import com.linecorp.armeria.common.HttpResponse;
import com.linecorp.armeria.common.ResponseHeaders;

import reactor.core.publisher.Flux;
sb.service("/", (ctx, req) -> {
    Flux<Product> products = dbClient.fetchAll();
    Flux<HttpData> data = products.map(product -> HttpData.of(...));
    return HttpResponse.of(ResponseHeaders.of(200), data);
}
```

Reply:
That's a good point.

I thought about using API from Reactor or RxJava, but I have felt a bit reluctant because a user might not want to use those libraries.
But there's no harm to introduce it anyway, so how about explaining `HttpResponseWriter` first and then introducing another version that uses `Flux`?

Reply:
That makes sense. I was worried that users feel difficult to handle streams with Armeria.
Had a chat with @minwoox and @trustin and @trustin suggested explaining step by step how to stream data from small data to large data.